File: Day_28/Q109.cpp
#include <bits/stdc++.h>
using namespace std;

//Structure For Library Details
struct Library{
    int bookId;
    string bookName;
    string author;
};

//Add Book In Library
void addBook(vector<Library> &lib){

    Library b;

    cout << "Enter Book ID : ";
    cin >> b.bookId;

    cin.ignore();

    cout << "Enter Book Name : ";
    getline(cin,b.bookName);

    cout << "Enter Author Name : ";
    getline(cin,b.author);

    lib.push_back(b);

    cout << "Book Added Successfully\n";
}

//Display All Books
void displayBooks(vector<Library> lib){

    if(lib.size() == 0){
        cout << "No Books Available\n";
        return;
    }

    for(int i = 0;i < lib.size();i++){

        cout << "\nBook ID : " << lib[i].bookId << endl;
        cout << "Book Name : " << lib[i].bookName << endl;
        cout << "Author : " << lib[i].author << endl;
    }
}

//Search Book By Book ID
void searchBook(vector<Library> lib){

    int id;
    bool found = false;

    cout << "Enter Book ID : ";
    cin >> id;

    for(int i = 0;i < lib.size();i++){

        if(lib[i].bookId == id){

            cout << "\nBook Found\n";
            cout << "Book Name : " << lib[i].bookName << endl;
            cout << "Author : " << lib[i].author << endl;

            found = true;
        }
    }

    if(found == false) cout << "Book Not Found\n";
}

int main(){

    vector<Library> lib;
    int choice;

    do{

        //Menu Driven Program
        cout << "\n1. Add Book";
        cout << "\n2. Display Books";
        cout << "\n3. Search Book";
        cout << "\n4. Exit";
        cout << "\nEnter Choice : ";
        cin >> choice;

        if(choice == 1) addBook(lib);

        else if(choice == 2) displayBooks(lib);

        else if(choice == 3) searchBook(lib);

    }while(choice != 4);

    return 0;
}

File: Day_28/Q109.java
import java.util.*;

//Class For Library Details
class Library{
    int bookId;
    String bookName;
    String author;
}

class Q109{

    static Scanner sc = new Scanner(System.in);

    //Add Book In Library
    static void addBook(ArrayList<Library> lib){

        Library b = new Library();

        System.out.print("Enter Book ID : ");
        b.bookId = sc.nextInt();

        sc.nextLine();

        System.out.print("Enter Book Name : ");
        b.bookName = sc.nextLine();

        System.out.print("Enter Author Name : ");
        b.author = sc.nextLine();

        lib.add(b);

        System.out.println("Book Added Successfully");
    }

    //Display All Books
    static void displayBooks(ArrayList<Library> lib){

        if(lib.size() == 0){
            System.out.println("No Books Available");
            return;
        }

        for(int i = 0;i < lib.size();i++){

            System.out.println("\nBook ID : " + lib.get(i).bookId);
            System.out.println("Book Name : " + lib.get(i).bookName);
            System.out.println("Author : " + lib.get(i).author);
        }
    }

    //Search Book By Book ID
    static void searchBook(ArrayList<Library> lib){

        int id;
        boolean found = false;

        System.out.print("Enter Book ID : ");
        id = sc.nextInt();

        for(int i = 0;i < lib.size();i++){

            if(lib.get(i).bookId == id){

                System.out.println("\nBook Found");
                System.out.println("Book Name : " + lib.get(i).bookName);
                System.out.println("Author : " + lib.get(i).author);

                found = true;
            }
        }

        if(found == false) System.out.println("Book Not Found");
    }

    public static void main(String args[]){

        ArrayList<Library> lib = new ArrayList<>();

        int choice;

        do{

            //Menu Driven Program
            System.out.println("\n1. Add Book");
            System.out.println("2. Display Books");
            System.out.println("3. Search Book");
            System.out.println("4. Exit");

            System.out.print("Enter Choice : ");
            choice = sc.nextInt();

            if(choice == 1) addBook(lib);

            else if(choice == 2) displayBooks(lib);

            else if(choice == 3) searchBook(lib);

        }while(choice != 4);

        sc.close();
    }
}
